### ISSUE TITLE
Cannot find executable for ideviceinfo

### ISSUE THREAD

**Issue by lczsf** [flutter/flutter#43398]
From URL: https://www.flutterchina.club/get-started/install.md

**Comment by lczsf** [comment]
Oops; flutter has exited unexpectedly.
Unhandled exception:
Invalid argument(s): Cannot find executable for /Users/chaolee/development/flutter/bin/cache/artifacts/libimobiledevice/ideviceinfo.

**Comment by iapicca** [comment]
Hi @lczsf 
can you please provide your updated  `flutter doctor -v`?

Also, to better address the issue, would be helpful
if you could post a self contained app on github
or the steps to reproduce it.

Lastly if the issue is related to the package 
you might want to take a look to this possible [solution](https://github.com/flutter/flutter/issues/24600).

Thank you

**Comment by ammichael** [comment]
I also have this issue. Here's my `flutter doctor -v`:

```
✓] Flutter (Channel stable, v1.9.1+hotfix.6, on Mac OS X 10.15.1 19B88, locale en-BR)
    • Flutter version 1.9.1+hotfix.6 at /Users/mike/flutter
    • Framework revision 68587a0916 (7 weeks ago), 2019-09-13 19:46:58 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

⣽
Sending crash report to Google.
⣯Crash report sent (report ID: 39d75b496bae3ea1)
Oops; flutter has exited unexpectedly.
⣷Unhandled exception:
Invalid argument(s): Cannot find executable for /Users/mike/flutter/bin/cache/artifacts/libimobiledevice/ideviceinfo.
#0      _getExecutable (package:process/src/interface/local_process_manager.dart:127:5)
#1      LocalProcessManager.run (package:process/src/interface/local_process_manager.dart:68:30)
#2      IMobileDevice.getInfoForDevice (package:flutter_tools/src/ios/mac.dart:199:57)
<asynchronous suspension>
#3      IOSDevice.getAttachedDevices (package:flutter_tools/src/ios/devices.dart:184:55)
<asynchronous suspension>
#4      IOSDevices.pollingGetDevices (package:flutter_tools/src/ios/devices.dart:117:57)
#5      PollingDeviceDiscovery.devices (package:flutter_tools/src/device.dart:270:52)
<asynchronous suspension>
#6      DeviceManager.getAllConnectedDevices (package:flutter_tools/src/device.dart:136:46)
<asynchronous suspension>
#7      DeviceValidator.validate (package:flutter_tools/src/doctor.dart:760:54)
<asynchronous suspension>
#8      Doctor.startValidatorTasks (package:flutter_tools/src/doctor.dart:140:52)
#9      Doctor.diagnose (package:flutter_tools/src/doctor.dart:211:41)
#10     _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:43:6)
#11     Doctor.diagnose (package:flutter_tools/src/doctor.dart:201:24)
#12     _doctorText.<anonymous closure> (package:flutter_tools/runner.dart:202:26)
#13     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:154:29)
#14     _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:43:6)
#15     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:154:7)
#16     _rootRun (dart:async/zone.dart:1124:13)
#17     _CustomZone.run (dart:async/zone.dart:1021:19)
#18     _runZoned (dart:async/zone.dart:1516:10)
#19     runZoned (dart:async/zone.dart:1463:12)
#20     AppContext.run (package:flutter_tools/src/base/context.dart:153:18)
#21     _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:43:6)
#22     AppContext.run (package:flutter_tools/src/base/context.dart:140:19)
#23     _doctorText (package:flutter_tools/runner.dart:201:19)
#24     _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:43:6)
#25     _doctorText (package:flutter_tools/runner.dart:197:27)
#26     _createLocalCrashReport (package:flutter_tools/runner.dart:179:32)
#27     _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:43:6)
#28     _createLocalCrashReport (package:flutter_tools/runner.dart:164:37)
#29     _handleToolError (package:flutter_tools/runner.dart:134:33)
#30     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#31     _rootRunUnary (dart:async/zone.dart:1132:38)
#32     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#33     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#34     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#35     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#36     Future._completeWithValue (dart:async/future_impl.dart:522:5)
#37     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:30:15)
#38     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:288:13)
#39     CrashReportSender.sendReport (package:flutter_tools/src/reporting/crash_reporting.dart)
#40     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#41     _rootRunUnary (dart:async/zone.dart:1132:38)
#42     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#43     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#44     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#45     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#46     Future._complete (dart:async/future_impl.dart:512:7)
#47     Stream.fold.<anonymous closure> (dart:async/stream.dart:801:18)
#48     _rootRun (dart:async/zone.dart:1120:38)
#49     _CustomZone.run (dart:async/zone.dart:1021:19)
#50     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#51     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:389:13)
#52     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:399:15)
#53     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:283:7)
#54     _SinkTransformerStreamSubscription._close (dart:async/stream_transformers.dart:96:11)
#55     _EventSinkWrapper.close (dart:async/stream_transformers.dart:23:11)
#56     _StringAdapterSink.close (dart:convert/string_conversion.dart:249:11)
#57     _Utf8ConversionSink.close (dart:convert/string_conversion.dart:300:20)
#58     _ConverterStreamEventSink.close (dart:convert/chunked_conversion.dart:80:18)
#59     _SinkTransformerStreamSubscription._handleDone (dart:async/stream_transformers.dart:141:24)
#60     _rootRun (dart:async/zone.dart:1120:38)
#61     _CustomZone.run (dart:async/zone.dart:1021:19)
#62     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#63     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:389:13)
#64     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:399:15)
#65     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:283:7)
#66     _ForwardingStream._handleDone (dart:async/stream_pipe.dart:106:10)
#67     _ForwardingStreamSubscription._handleDone (dart:async/stream_pipe.dart:172:13)
#68     _rootRun (dart:async/zone.dart:1120:38)
#69     _CustomZone.run (dart:async/zone.dart:1021:19)
#70     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#71     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:389:13)
#72     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:399:15)
#73     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:283:7)
#74     _SinkTransformerStreamSubscription._close (dart:async/stream_transformers.dart:96:11)
#75     _EventSinkWrapper.close (dart:async/stream_transformers.dart:23:11)
#76     _Uint8ListConversionSink.close (dart:_http/http_impl.dart:577:13)
#77     _ConverterStreamEventSink.close (dart:convert/chunked_conversion.dart:80:18)
#78     _SinkTransformerStreamSubscription._handleDone (dart:async/stream_transformers.dart:141:24)
#79     _rootRun (dart:async/zone.dart:1120:38)
#80     _CustomZone.run (dart:async/zone.dart:1021:19)
#81     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#82     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:389:13)
#83     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:399:15)
#84     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:283:7)
#85     _SinkTransformerStreamSubscription._close (dart:async/stream_transformers.dart:96:11)
#86     _EventSinkWrapper.close (dart:async/stream_transformers.dart:23:11)
#87     _ByteAdapterSink.close (dart:convert/byte_conversion.dart:62:11)
#88     _FilterSink.close (dart:io/data_transformer.dart:637:11)
#89     _ConverterStreamEventSink.close (dart:convert/chunked_conversion.dart:80:18)
#90     _SinkTransformerStreamSubscription._handleDone (dart:async/stream_transformers.dart:141:24)
#91     _rootRun (dart:async/zone.dart:1120:38)
#92     _CustomZone.run (dart:async/zone.dart:1021:19)
#93     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#94     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:389:13)
#95     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:399:15)
#96     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:283:7)
#97     _ForwardingStream._handleDone (dart:async/stream_pipe.dart:106:10)
#98     _ForwardingStreamSubscription._handleDone (dart:async/stream_pipe.dart:172:13)
#99     _rootRun (dart:async/zone.dart:1120:38)
#100    _CustomZone.run (dart:async/zone.dart:1021:19)
#101    _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#102    _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:389:13)
#103    _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:399:15)
#104    _BufferingStreamSubscription._close (dart:async/stream_impl.dart:283:7)
#105    _SyncStreamControllerDispatch._sendDone (dart:async/stream_controller.dart:772:19)
#106    _StreamController._closeUnchecked (dart:async/stream_controller.dart:629:7)
#107    _StreamController.close (dart:async/stream_controller.dart:622:5)
#108    _HttpParser._closeIncoming (dart:_http/http_parser.dart:1037:23)
#109    _HttpParser._doParse (dart:_http/http_parser.dart:752:11)
#110    _HttpParser._parse (dart:_http/http_parser.dart:318:7)
#111    _HttpParser._onData (dart:_http/http_parser.dart:810:5)
#112    _rootRunUnary (dart:async/zone.dart:1132:38)
#113    _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#114    _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#115    _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#116    _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#117    _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:764:19)
#118    _StreamController._add (dart:async/stream_controller.dart:640:7)
#119    _StreamController.add (dart:async/stream_controller.dart:586:5)
#120    _Socket._onData (dart:io-patch/socket_patch.dart:1791:41)
#121    _rootRunUnary (dart:async/zone.dart:1132:38)
#122    _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#123    _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#124    _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#125    _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#126    _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:764:19)
#127    _StreamController._add (dart:async/stream_controller.dart:640:7)
#128    _StreamController.add (dart:async/stream_controller.dart:586:5)
#129    _RawSecureSocket._sendReadEvent (dart:io/secure_socket.dart:1004:19)
#130    _rootRun (dart:async/zone.dart:1120:38)
#131    _CustomZone.run (dart:async/zone.dart:1021:19)
#132    _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#133    _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#134    _rootRun (dart:async/zone.dart:1124:13)
#135    _CustomZone.run (dart:async/zone.dart:1021:19)
#136    _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:947:23)
#137    Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:21:15)
#138    _Timer._runTimers (dart:isolate-patch/timer_impl.dart:382:19)
#139    _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:416:5)
#140    _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:12)
```

**Comment by sameeratennakoon** [comment]
Go to the project directory in the terminal and run `pod setup` . This fixed the same issue that I had.

**Comment by jmagman** [comment]
Another case of this at https://github.com/flutter/flutter/issues/44549#issuecomment-553353575

**Comment by jmagman** [comment]
\cc @christopherfujino

**Comment by nikitakuhta** [comment]
`pod setup` not solved issue, but when I removed all cached binaries of flutter, everything become fine.
`rm -rf ~/dev/flutter/bin/cache`

**Comment by christopherfujino** [comment]
The PR that closed this was reverted, but should be re-landed in #50252

**Comment by ride4sun** [comment]
Just installed a new Mac - still see the same issue. Please reopen

**Comment by christopherfujino** [comment]
@ride4sun what flutter version are you using (can you provide full `flutter doctor -v` output)? Note, that the fix for this only landed on version 1.15.2, which is currently only on master and dev channel.

**Comment by ride4sun** [comment]
Sure:
jay@mini test_2 % flutter doctor -v
[✓] Flutter (Channel stable, v1.12.13+hotfix.8, on Mac OS X 10.15.3 19D76,
    locale en-US)
    • Flutter version 1.12.13+hotfix.8 at /Users/jay/development/flutter
    • Framework revision 0b8abb4724 (3 weeks ago), 2020-02-11 11:44:36 -0800
    • Engine revision e1e6ced81d
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/jay/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.9.0

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)

[✓] Connected device (1 available)
    • Jay’s iPhone • bc987fccd61b08a374d7fd1329958f428155b60b • ios • iOS 12.1.4

• No issues found!

**Comment by christopherfujino** [comment]
@ride4sun the permanent fix for this only landed in version 1.15.2, which you can get by switching to the dev channel (e.g. `flutter channel dev`). Alternatively, to fix this state locally, you can `rm -rf /Users/jay/development/flutter/bin/cache && flutter doctor -v`, which will wipe out your local cache of binaries, and then invoking the doctor should cause a re-download of necessary binaries.
